Documentation, galaxy wrappers and bug fixes

joachimwolff opened the pull request:
This PR adds:
- Bug fix in hicBuildMatrix concerning danglingSequnce and restrictionSequence
- Using now system environment variable to set bam buffer dir: HICEXPLORER_FILE_BUFFER_DIR
- Update of the mESC tutorial
- Update of doc-string where necessary
- Update of the galaxy wrapper, including a rewrite of hicPlotTADs.xml. Please review hicPlotTADs.xml if I have all possible parameters for the track file included. I reorganized the wrapper in categories and I display in each category only the options that are useful for it.
- Extension of hicFindTADs with fdr and the option to chose no multiple correction
- Merged the .travis.yml file of bgruening, see PR #115. It is now possible to test new source code with the galaxy wrappers to see if they are working or if they need an update. The wrappers should NOT be used as regular test cases! For this we have testcases i.e. nosetests. 

For some reason the wrapper test of hicBuildMatrix fails, and I am running out of ideas how to fix it. Maybe someone else knows something, at [travis](https://travis-ci.org/joachimwolff/HiCExplorer/builds) you can have a look at my many tries, perhaps you see something I have missed.  

Please review it carefully if there are mistakes e.g. in the mESC tutorial and do additional testing to ensure no more major bugs are in our software!

Thanks a lot.

bgruening commented:
I think hicFindEnrichedContacts.xml was not commited and is missing from this PR.

joachimwolff commented:
Looks like hicFindEnrichedContacts.xml is empty. It was created 2 years ago but never filled with content. Should I create a wrapper for it or leave it this way because no one really missed it?

bgruening commented:
Probably not need for this release but then it should be removed from the tests.

joachimwolff commented:
I removed the galaxy wrappers, removed the usage in hicBuildMatrix.py. It is now a bit slower but I think it is easier this way for our users. I have some improvements for this on my mind, but they will not make it to this release.

fidelram commented on hicexplorer/hicFindTADs.py:
```diff
@@ -1190,10 +1201,24 @@ def min_pvalue(self, min_idx):
-        pvalues[np.array([e > 1 if ~np.isnan(e) else False for e in pvalues])] = 1
+
+        # fdr
+        if self.multipleComparisons == 'fdr':
```
did you compare the resulting fdr values with the result from R just to be sure that there are no problems?

joachimwolff replied:
I did not so far and will do it in the next days. I used https://matthew-brett.github.io/teaching/fdr.html as a reference.

joachimwolff replied:
I am close, but not close enough :( I will debug it and get back to you.

|q-value|R fdr | my fdr|
|----------|-------|---------|
|0.01     |256  |253     |               
|0.02     |335  |329     |
|0.03     |388  |367     |
|0.04     |413  |403     |
|0.05     |425  |414     |
|0.1       |489  |470     |
|0.2       |537  |515     |

joachimwolff replied:
If only the fdr is considered the numbers are equal. The difference and the more restrictive behaviour comes from the check against the membership of `delta_of_min` [Line 911](https://github.com/joachimwolff/HiCExplorer/blob/468608a2d4372e7cf97afd6f33f1014538779be5/hicexplorer/hicFindTADs.py#L911) and the subsequent test `if delta_of_min[idx] >= self.delta` [Line 914](https://github.com/joachimwolff/HiCExplorer/blob/468608a2d4372e7cf97afd6f33f1014538779be5/hicexplorer/hicFindTADs.py#L914).
I think this test should be included for the fdr too. Nevertheless, I discovered a few potential bugs in this and changed it in PR #122

fidelram commented on hicexplorer/hicInfo.py:
```diff
@@ -10,8 +10,8 @@ def parse_arguments(args=None):
-                        help='matrices to add. Must have the same shape.',
-                        metavar='.h5 of .npz file format',
+                        help='The matrix (matrices) to get information about.',
+                        metavar='HiCExplorer supports: h5, npz, dekker and lieberman file format.',
```
Here we have to be careful. What I call dekker format are the matrices produced by Job Dekker about 2-3 years ago. I don't know what are they using now. For Lieberman, we mean the matrices produced in the Lieberman publications from 2015.

joachimwolff replied:
Should we set the formats `dekker` and `lieberman` to `deprecated`?

bgruening replied:
I would deprecate this and go forward with Cooler.

fidelram replied:
I know that people use that. I would not deprecate it at the moment, just point out that this refers to the format used by Job Dekker and Erez Lieberman Aiden and that they may change in the future.

bgruening replied:
@fidelram I think we should indicate that this functionality will be removed at some point. It will simply the code a lot, especially with the cooler format that will be coming soon. I would propose an additional tool that converts from X to cooler. With such a tool people can still use the older formats.

fidelram replied:
I will merge and take care of this afterwards.

fidelram commented on hicexplorer/hicMergeMatrixBins.py:
```diff
@@ -154,7 +154,7 @@ def merge_bins(hic, num_bins):
     ----------
 
-    hic : HiCMatrix object
+    hic : Hi-C Matrix object
```
Here we don't need to replace by Hi-C Matrix because the name of the class is HiCMatrix.

fidelram commented on hicexplorer/hicMergeTADbins.py:
```diff
@@ -36,11 +36,11 @@ def parse_arguments():
-    Reduces the HiC matrix by merging the counts of tad bins.
-    :param hic: hicMatrix object
+    Reduces the Hi-C matrix by merging the counts of tad bins.
+    :param hic: Hi-C matrix object
```
Same as the previous comment

fidelram commented on hicexplorer/hicMergeTADbins.py:
```diff
@@ -36,11 +36,11 @@ def parse_arguments():
+    :param hic: Hi-C matrix object
     :param boundary_id_list list of tad boundary bin ids
     :param filename Name to save the resulting matrix
-    :return: hicMatrix object
```
Same as before. Although should be `HiCMatrix` with upper case H

fidelram commented on hicexplorer/hicMergeTADbins.py:
```diff
@@ -87,7 +87,7 @@ def merge_tad_bins(hic, boundary_id_list, filename):
 def get_boundary_bin_id(hic, bed_fh):
     """
-    :param hic: HiCMatrix object
+    :param hic: Hi-C matrix object
```
same

fidelram commented on hicexplorer/hicPlotDistVsCounts.py:
```diff
@@ -100,7 +100,7 @@ def compute_distance_mean(hicmat, maxdepth=None, perchr=False):
     Parameters
     ----------
-    hicmat: HiCMatrix object
+    hicmat: Hi-C matrix object
```
Same

joachimwolff commented on hicexplorer/hicFindTADs.py:
```diff
@@ -345,22 +342,22 @@ def compute_matrix(bins_list, min_win_size=8, max_win_size=50, step_len=2):
 
     def __init__(self, matrix, num_processors=1, max_depth=None, min_depth=None, step=None, delta=0.01,
-                 pvalue=0.01, min_boundary_distance=None, use_zscore=True):
+                 min_boundary_distance=None, use_zscore=True, pMultipleComparisons="fdr", pThresholdComparisons=0.01):
```
I think we should use one terminology for the public interface. Camel case or "_" but we should not mix them.

bgruening commented:
Wuhu, merged! Thanks a lot @fidelram!
